scanomatic/image_analysis/grayscale_detection.py

```diff
@@ -168,39 +169,41 @@ def get_para_trimmed_slice(
     return im_ortho_trimmed
 
 
-def get_grayscale(
-    fixture,  # FixtureImage (but circular import)
-    grayscale_area_model: GrayScaleAreaModel,
-    debug: bool = False,
-) -> tuple[Optional[np.ndarray], Optional[list[float]]]:
-    im = fixture.get_grayscale_im_section(grayscale_area_model)
-    if im is None:
-        return None, None
-    return get_grayscale_image_analysis(
-        im,
-        grayscale_area_model.name,
-        debug=debug,
-    )
+def get_grayscale_im_section(
+    im: np.ndarray,
+    grayscale_model: GrayScaleAreaModel,
+    scale=1.0,
+) -> np.ndarray:
+    try:
+        return im[
+            int(round(max(grayscale_model.y1 * scale, 0))):
+            int(round(min(grayscale_model.y2 * scale, im.shape[0]))),
+            int(round(max(grayscale_model.x1 * scale, 0))):
+            int(round(min(grayscale_model.x2 * scale, im.shape[1])))
+        ]
+    except (IndexError, TypeError):
+        msg = "Could not extract grayscale section"
+        raise GrayscaleError(msg)
 
 
-def get_grayscale_image_analysis(
-    im: np.ndarray,
-    grayscale_name: str,
+def detect_grayscale(
```

Comment:
Since this can now raise the calibration_api and data_api that seem to be the sources of the request should probably catch that exception and return a useful error